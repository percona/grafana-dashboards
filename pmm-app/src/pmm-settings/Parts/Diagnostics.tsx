import React from 'react';
import {Button, Col, Row} from 'antd';

const Diagnostics = props => {
  return (
    <>
      <Row style={{ color: 'white', marginBottom: '10px' }} align={'middle'}>
        <Col span={12} style={{ display: 'flex', justifyContent: 'flex-start', paddingRight: '20px' }}>
          <span>We will have a box for troubleshooting - how to download logs (PMM Server logs.zip)</span>
        </Col>
      </Row>
      <Row style={{ color: 'white', marginBottom: '10px' }} align={'middle'}>
        <Col span={12} style={{ display: 'flex', justifyContent: 'flex-start', paddingRight: '20px' }}>
          <Button icon={'download'}>Download PMM Server Logs</Button>
        </Col>
      </Row>
    </>
  );
};

export default Diagnostics;
