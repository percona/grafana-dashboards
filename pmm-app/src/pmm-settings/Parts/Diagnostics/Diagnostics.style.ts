import { css } from 'emotion';

export const styles = {
  row: css`
    color: white;
    margin-botton: 10px;
  `,
  column: css`
    display: flex;
    justify-content: flex-start;
    padding-right: 20px;
  `,
};
