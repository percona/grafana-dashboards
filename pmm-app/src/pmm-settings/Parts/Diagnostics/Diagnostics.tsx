import React from 'react';
import { Button, Col, Row } from 'antd';

const Diagnostics = () => {
  return (
    <>
      <Row style={{ color: 'white', marginBottom: '10px' }} align="middle">
        <Col span={24}>
          You can download server logs to make the problem detection simpler. Please include this file if you
          are submitting a bug report.
        </Col>
      </Row>
      <Row style={{ color: 'white', marginBottom: '10px' }} align="middle">
        <Col span={12} style={{ display: 'flex', justifyContent: 'flex-start', paddingRight: '20px' }}>
          <Button icon="download" href="/logs.zip">
            Download PMM Server Logs
          </Button>
        </Col>
      </Row>
    </>
  );
};

export default Diagnostics;
