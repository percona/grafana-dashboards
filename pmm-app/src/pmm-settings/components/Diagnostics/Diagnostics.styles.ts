import { stylesFactory, selectThemeVariant } from '@grafana/ui';
import { GrafanaTheme } from '@grafana/data';
import { css } from 'emotion';

export const getStyles = stylesFactory((theme: GrafanaTheme) => {
  const { colors }: any = theme;
  const background = selectThemeVariant(
    {
      light: `linear-gradient(180deg, ${colors.gray98} 0%, #e7eaf0 100%)`,
      dark: `linear-gradient(180deg, ${colors.gray15} 0%, #292c31 100%)`
    },
    theme.type,
  );
  const hoverBackground = selectThemeVariant(
    { light: colors.gray98, dark: colors.gray15 },
    theme.type,
  );
  const border = selectThemeVariant(
    { light: colors.gray85, dark: colors.gray25 },
    theme.type,
  );
  const textColor = selectThemeVariant(
    { light: colors.text, dark: colors.white },
    theme.type,
  );

  return {
    diagnosticsWrapper: css`
      align-items: flex-end;
      display: flex;
      flex: 1;
      flex-direction: column;
    `,
    diagnosticsLabel: css`
      display: flex;
      i {
        margin-left: ${theme.spacing.xs};
      }
    `,
    diagnosticsButton: css`
      background: ${background};
      border: ${theme.border.width.sm} solid ${border};
      color: ${textColor};
      font-size: ${theme.typography.size.sm};
      margin-top: ${theme.spacing.md};
      &:hover {
        background: ${hoverBackground};
        color: ${textColor};
      }
      &:focus {
        background: ${hoverBackground};
      }
      span {
        align-items: flex-end;
        display: flex;
      }
      svg {
        margin-right: ${theme.spacing.sm};
        fill: ${textColor};
      }
    `
  };
});
