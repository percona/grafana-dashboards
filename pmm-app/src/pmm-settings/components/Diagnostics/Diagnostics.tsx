import React, { FC } from 'react';
import {
  Icon,
  LinkButton,
  Tooltip,
  useTheme
} from '@grafana/ui';
import { Messages } from 'pmm-settings/Settings.messages';
import { getSettingsStyles } from 'pmm-settings/Settings.styles';
import { Download } from 'shared/components/Elements/Icons/Download';
import { getStyles } from './Diagnostics.styles';

export const Diagnostics: FC = () => {
  const theme = useTheme();
  const styles = getStyles(theme);
  const settingsStyles = getSettingsStyles(theme);
  const { diagnostics: { action, label, tooltip } } = Messages;

  return (
    <div className={styles.diagnosticsWrapper}>
      <div
        className={settingsStyles.labelWrapper}
        data-qa="diagnostics-label"
      >
        {label}
        <Tooltip content={tooltip}>
          <div>
            <Icon name="info-circle" />
          </div>
        </Tooltip>
      </div>
      <LinkButton
        href="/logs.zip"
        className={styles.diagnosticsButton}
        data-qa="diagnostics-button"
      >
        <Download />
        <span>{action}</span>
      </LinkButton>
    </div>
  );
};
