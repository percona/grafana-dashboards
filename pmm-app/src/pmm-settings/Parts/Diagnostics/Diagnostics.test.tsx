// Just a stub test
import Diagnostics from './Diagnostics';
import React from 'react';
import renderer from 'react-test-renderer';

describe('Diagnostics part test', () => {
  it('Renders correct', () => {
    const component = renderer.create(<Diagnostics />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
