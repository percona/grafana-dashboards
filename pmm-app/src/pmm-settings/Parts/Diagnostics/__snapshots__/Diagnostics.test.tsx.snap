// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Diagnostics part test Renders correct 1`] = `
<div
  className="ant-row"
  style={
    Object {
      "color": "white",
      "marginBottom": "10px",
    }
  }
>
  <div
    className="ant-col ant-col-12"
    style={
      Object {
        "display": "flex",
        "justifyContent": "flex-start",
        "paddingRight": "20px",
      }
    }
  >
    <a
      className="ant-btn"
      href="/logs.zip"
      onClick={[Function]}
    >
      <i
        aria-label="icon: download"
        className="anticon anticon-download"
      >
        <svg
          aria-hidden="true"
          className=""
          data-icon="download"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M505.7 661a8 8 0 0 0 12.6 0l112-141.7c4.1-5.2.4-12.9-6.3-12.9h-74.1V168c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v338.3H400c-6.7 0-10.4 7.7-6.3 12.9l112 141.8zM878 626h-60c-4.4 0-8 3.6-8 8v154H214V634c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v198c0 17.7 14.3 32 32 32h684c17.7 0 32-14.3 32-32V634c0-4.4-3.6-8-8-8z"
          />
        </svg>
      </i>
      <span>
        Download PMM Server Logs
      </span>
    </a>
  </div>
</div>
`;
